applications/StructuralMechanicsApplication/custom_elements/truss_elements/linear_truss_element_2D.cpp

```diff
@@ -644,6 +644,54 @@ void LinearTrussElement2D<TNNodes>::CalculateOnIntegrationPoints(
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<SizeType TNNodes>
+void LinearTrussElement2D<TNNodes>::CalculateOnIntegrationPoints(
+    const Variable<Vector>& rVariable,
+    std::vector<Vector>& rOutput,
+    const ProcessInfo& rProcessInfo
+    )
+{
+    const auto& integration_points = IntegrationPoints(GetIntegrationMethod());
+    rOutput.resize(integration_points.size());
+
+    if (rVariable == PK2_STRESS_VECTOR) {
+        ConstitutiveLaw::Parameters cl_values(GetGeometry(), GetProperties(), rProcessInfo);
+        cl_values.GetOptions().Set(ConstitutiveLaw::COMPUTE_STRESS             , true);
+        cl_values.GetOptions().Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, true);
+
+        const double length = CalculateLength();
+
+        // Let's initialize the cl values
+        VectorType strain_vector(1), stress_vector(1);
+        MatrixType constitutive_matrix(1,1);
+        strain_vector.clear();
+        cl_values.SetStrainVector(strain_vector);
+        cl_values.SetStressVector(stress_vector);
+        cl_values.SetConstitutiveMatrix(constitutive_matrix);
+        SystemSizeBoundedArrayType nodal_values(SystemSize);
+        GetNodalValuesVector(nodal_values);
+
+        SystemSizeBoundedArrayType dN_dX;
+
+        // Loop over the integration points
+        for (SizeType integration_point = 0; integration_point < integration_points.size(); ++integration_point) {
+            GetFirstDerivativesShapeFunctionsValues(dN_dX, length, integration_points[integration_point].X());
+
+            strain_vector[0] = inner_prod(dN_dX, nodal_values);
+
+            mConstitutiveLawVector[integration_point]->CalculateMaterialResponsePK2(cl_values);
+            auto stress = cl_values.GetStressVector()[0];
+            if (GetProperties().Has(TRUSS_PRESTRESS_PK2)) {
+                stress += GetProperties()[TRUSS_PRESTRESS_PK2];
+            }
+
+            rOutput[integration_point] = ScalarVector(1, stress);
```

Comment:
If this part of code is identical to another part in this class, we could extract a function for the common functionality

Reply:
A few lines are extracted as a function.